### PR TITLE
LPS-136442 Refactor version comparison to compare with latest production version

### PR DESCRIPTION
**Pull request by stsquared99**
@Preston-Crary I just didn't set the correct threads when fetching the content, in case other models need to be fetched as part of the getContent method. Also I simplified the errors when fetching content. I am matching the behavior of the data rendering. If there is always an error when fetching content, it just isn't available, similar to how the render defaults to the base display renderer.

### CODE REVIEW

**Comment by Preston-Crary** [comment]
ci:test:relevant

**Comment by Preston-Crary** [comment]
@stsquared99 did you check history view with add/modify/delete to make sure that part has correct thread locals too?

**Comment by stsquared99** [comment]
@Preston-Crary I did check. I had a random instance where I couldn't fetch the content on a published delete for journal, but that seems to have been a stale db and I haven't seen that happen again. I'll keep an eye out for it but it has since then always worked.

I did catch one typo on a ctid in the render code, so I fixed and pushed that just now.

**Comment by stsquared99** [comment]
ci:test:relevant

File: modules/apps/change-tracking/change-tracking-spi/src/main/java/com/liferay/change/tracking/spi/display/CTDisplayRenderer.java [hunk 1]
```diff
@@ -34,6 +34,10 @@
  */
 public interface CTDisplayRenderer<T> {
 
+	public default T fetchLatestModel(T model) {
```
[the remark this answers is not in the document]
Comment (stsquared99) [reply modules/apps/change-tracking/change-tracking-spi/src/main/java/com/liferay/change/tracking/spi/display/CTDisplayRenderer.java:38]: Didn't we talk about that with brian the first time around? we name the method with the actual thing that is fetched? it returns a Model, not a Version object

[the remark this answers is not in the document]
Comment (Preston-Crary) [reply modules/apps/change-tracking/change-tracking-spi/src/main/java/com/liferay/change/tracking/spi/display/CTDisplayRenderer.java:38]: Let's also apply this to something besides journal that has versions, maybe DLFileVersion?

[the remark this answers is not in the document]
Comment (Preston-Crary) [reply modules/apps/change-tracking/change-tracking-spi/src/main/java/com/liferay/change/tracking/spi/display/CTDisplayRenderer.java:38]: That's fair, let's call it `fetchLatestVersionedModel()` like before then.

**Comment by Preston-Crary** [comment]
Looks good, just want to clarify that the method is for versions in the name and use it for something else beside journal. That way we can be more sure it will work for other use cases.

**Comment by stsquared99** [comment]
ci:stop

**Comment by stsquared99** [comment]
@Preston-Crary Sounds good. I will rename the method to that and also implement the methods for document version

**Comment by stsquared99** [comment]
Resent: https://github.com/Preston-Crary/liferay-portal/pull/1448

**Pull request closed without merging**